src/cim/components/modals/AddBmcHostYamlForm.tsx

```diff
@@ -0,0 +1,208 @@
+    saveAs(blob, 'bmc_hosts_template.yaml');
+  };
+
+  const importBMAsYAML = async () => {
+    setTextOpenFile('Opening file');
+    const input = document.createElement('input');
+    input.setAttribute('id', 'importBMAsYaml');
+    input.type = 'file';
+    input.accept = '.yaml';
+    input.style.visibility = 'hidden';
+    const body = document.querySelector('body');
+    body?.appendChild(input);
+    input.onchange = (e: Event) => {
+      input.remove();
+      const target = e.target as HTMLInputElement;
+      const file: File = (target.files as FileList)[0];
+      let maxSize = 2147483648; //2Gb
+      maxSize = 529145600; //600 Mb per provar
+      if (file.size > maxSize) {
+        setShowOpenFileButton(false);
+        setFileName(file.name);
+        setFileError('The file is too big. Upload a file up to 2 GB.');
+        setYamlContent([]);
+      } else {
+        const reader = new FileReader();
+        reader.readAsText(file, 'UTF-8');
+        reader.onload = async (readerEvent) => {
+          const content = readerEvent.target?.result;
+          // parse yaml
+          if (typeof content === 'string') {
+            await new Promise((r) => setTimeout(r, 100));
+            let yamlContent;
+            try {
+              yamlContent = jsYaml.loadAll(content);
+              setTextOpenFile(t('ai:Open file'));
+              setShowOpenFileButton(false);
+              setFileName(file.name);
+              setFileError(undefined);
+              setYamlContent(yamlContent);
+            } catch (err) {
+              setTextOpenFile(t('ai:Open file'));
+              setShowOpenFileButton(false);
+              setFileName(file.name);
+              setFileError(
+                'File is not structured correctly. Use the template to use the right file structure.',
+              );
+              setYamlContent([]);
+            }
+          }
+        };
+      }
+    };
+    input.click();
+  };
+
+  const deleteFileContent = () => {
+    setFileError(undefined);
+    setFileName(undefined);
+    setShowOpenFileButton(true);
+    setTextOpenFile(t('ai:Open file'));
+  };
+
+  const handleSubmit: FormikConfig<AddYamlValues>['onSubmit'] = async () => {
+    try {
+      setError(undefined);
+      if (yamlContent !== undefined) {
+        await onCreateBmcByYaml(yamlContent);
+      }
+
+      onClose();
+    } catch (e) {
+      setError(getErrorMessage(e));
+    }
+  };
+
+  return (
+    <Formik initialValues={emptyValues(t)} isInitialValid={false} onSubmit={handleSubmit}>
+      {({ submitForm }: FormikProps<AddYamlValues>) => (
+        <ModalBoxBody>
+          <EmptyState>
+            <Title headingLevel="h4" size="lg">
+              <div>
+                <p>
+                  <UploadIcon size="xl" color="grey" />
+                </p>
+                {t('ai:Upload a YAML file')}
+              </div>
+            </Title>
+            <EmptyStateBody>
+              <div>
```

Comment:
there is no spacing between elements
please use `Stack` and `StackItem` with gutter https://patternfly-react.surge.sh/layouts/stack#with-gutter